ArbFileType.js

```diff
@@ -0,0 +1,217 @@
+var ArbFile = require("./ArbFile.js");
+
+var ArbFileType = function(project) {
+    this.type = "arb";
+    this.resourceType = "arb";
+    this.project = project;
+    this.extensions = [".arb"];
+    this.datatype = "arb";
+    this.names = ["intl_messages"];
+
+    this.API = project.getAPI();
+    this.logger = this.API.getLogger("loctool.plugin.ArbFileType");
+    this.extracted = this.API.newTranslationSet(project.getSourceLocale());
+    this.newres = this.API.newTranslationSet(project.getSourceLocale());
+    this.pseudo = this.API.newTranslationSet(project.getSourceLocale());
+
+    // generate all the pseudo bundles we'll need
+    if (project.pseudoLocale && Array.isArray(project.pseudoLocale)) {
+        this.pseudos = {};
+        project.pseudoLocale && project.pseudoLocale.forEach(function(locale) {
+            var pseudo = this.API.getPseudoBundle(locale, this, project);
+            if (pseudo) {
+                this.pseudos[locale] = pseudo;
+            }
+        }.bind(this));
+    }
+    if (project.pseudoLocales && typeof project.pseudoLocales == 'object') {
+        this.pseudos = {};
+        for (locale in project.pseudoLocales) {
+            var pseudo = this.API.getPseudoBundle(locale, this, project);
+            if (pseudo) {
+                this.pseudos[locale] = pseudo;
+            }
+        }
+    }
+};
+
+var defaultMappings = {
+    "**/intl_messages.arb": {
+        "template": "[dir]/intl_[localeUnder].arb"
+    }
+}
+/**
+ * Return the mapping corresponding to this path.
+ * @param {String} pathName the path to check
+ * @returns {Object} the mapping object corresponding to the
+ * path or undefined if none of the mappings match
+ */
+ArbFileType.prototype.getMappings = function(pathName) {
+    if (typeof(pathName) === "undefined") {
+        return undefined;
+    }
+    var mappings, match;
+    var arbMapSettings = this.project.settings && this.project.settings.arbMap;
+    if (arbMapSettings) {
+        mappings = arbMapSettings.mappings || defaultMappings;
+        var patterns = Object.keys(mappings);
+
+        if (patterns) {
+            match = patterns.find(function(pattern) {
+                return mm.isMatch(pathName, pattern);
+            });
+        }
+    }
+    return match && mappings[match];
+}
+
+ArbFileType.prototype.getDefaultMapping = function() {
+    return defaultMappings["**/intl_messages.arb"];
+}
+
+/**
+ * Return true if the given path is a appinfo.json or qcardinfo.json file and is handled
+ * by the current file type.
+ *
+ * @param {String} pathName path to the file being questions
+ * @returns {boolean} true if the path is a appinfo.json or qcardinfo.json file, or false
+ * otherwise
+ */
+ArbFileType.prototype.handles = function(pathName) {
+    this.logger.debug("ArbFileTyp handles " + pathName + "?");
+    if (!pathName) return false;
+    
+    var file = path.basename(pathName, ".arb");
+    ret = (this.names.indexOf(file) !== -1) ? true: false;
+    
```

Comment:
this doesn't check if the arb file is already localized. (ie. it has a name intl_<locale>.arb)

Reply:
This plugin only handle `intl_messages.arb` file. and the output file name is different. so I think i t doesn't need to check if the specific arb file is localized.

Reply:
I think you have it backwards. The loctool asks this function, "does this file need to be localized and therefore its strings need to be extracted and put into the new strings file."
 
Right now, if you call `ArbFileType.handles("foo/bar/intl_ja-JP.arb")` it will say "yes, localize it" whereas it should says, "no, it's an already localized file."

Reply:
In case  `ArbFileType.handles("foo/bar/intl_ja-JP.arb")` it will returns 'false`. because the arb file name doesn't match 'intl_messages.arb` . I think it doesn't need to filter  localized arb file among  other case of files. no...?
https://github.com/iLib-js/ilib-loctool-arb/blob/init/ArbFileType.js#L30